server/utils/utils.go

```diff
@@ -26,3 +32,157 @@ func GetLatestSchemaVersion(ctx context.Context, driver db.Driver, databaseName
+	migrationContext := base.NewMigrationContext()
+	migrationContext.InspectorConnectionConfig.Key.Hostname = config.host
+	port := 3306
+	if config.port != "" {
+		configPort, err := strconv.Atoi(config.port)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to convert port from string to int")
+		}
+		port = configPort
+	}
+	migrationContext.InspectorConnectionConfig.Key.Port = port
+	migrationContext.CliUser = config.user
+	migrationContext.CliPassword = config.password
+	migrationContext.DatabaseName = config.database
+	migrationContext.OriginalTableName = config.table
+	migrationContext.AlterStatement = statement
+	migrationContext.Noop = config.noop
+	migrationContext.ReplicaServerId = config.serverID
+	if config.isAWS {
+		migrationContext.AssumeRBR = true
+	}
+	// set defaults
+	migrationContext.AllowedRunningOnMaster = allowedRunningOnMaster
+	migrationContext.ConcurrentCountTableRows = concurrentCountTableRows
+	migrationContext.HooksStatusIntervalSec = hooksStatusIntervalSec
+	migrationContext.CutOverType = base.CutOverAtomic
+	migrationContext.ThrottleHTTPIntervalMillis = throttleHTTPIntervalMillis
+	migrationContext.ThrottleHTTPTimeoutMillis = throttleHTTPTimeoutMillis
+
+	if migrationContext.AlterStatement == "" {
+		return nil, errors.Errorf("alterStatement must be provided and must not be empty")
+	}
+	parser := ghostsql.NewParserFromAlterStatement(migrationContext.AlterStatement)
+	migrationContext.AlterStatementOptions = parser.GetAlterStatementOptions()
+
+	if migrationContext.DatabaseName == "" {
+		if !parser.HasExplicitSchema() {
+			return nil, errors.Errorf("database must be provided and database name must not be empty, or alterStatement must specify database name")
+		}
+		migrationContext.DatabaseName = parser.GetExplicitSchema()
+	}
+	if migrationContext.OriginalTableName == "" {
+		if !parser.HasExplicitTable() {
+			return nil, errors.Errorf("table must be provided and table name must not be empty, or alterStatement must specify table name")
+		}
+		migrationContext.OriginalTableName = parser.GetExplicitTable()
+	}
+	migrationContext.ServeSocketFile = config.socketFilename
+	migrationContext.PostponeCutOverFlagFile = config.postponeFlagFilename
+	migrationContext.TimestampAllTable = timestampAllTable
+	migrationContext.SetHeartbeatIntervalMilliseconds(heartbeatIntervalMilliseconds)
+	migrationContext.SetNiceRatio(niceRatio)
+	migrationContext.SetChunkSize(chunkSize)
+	migrationContext.SetDMLBatchSize(dmlBatchSize)
+	migrationContext.SetMaxLagMillisecondsThrottleThreshold(maxLagMillisecondsThrottleThreshold)
+	migrationContext.SetDefaultNumRetries(defaultNumRetries)
+	migrationContext.ApplyCredentials()
+	if err := migrationContext.SetCutOverLockTimeoutSeconds(cutoverLockTimeoutSeconds); err != nil {
+		return nil, err
+	}
+	if err := migrationContext.SetExponentialBackoffMaxInterval(exponentialBackoffMaxInterval); err != nil {
+		return nil, err
+	}
+	return migrationContext, nil
+}
```

Comment:
I think these lines belong to gh-ost task executor.

Reply:
As I need to move task executor to a new package parallel to task check package, these functions need to be shared. Let's keep it in util for now till we find a better place.